### PR TITLE
issue #348: vhost to server

### PR DESCRIPTION
**Pull request by renich**
Based on the discussion we had, I've renamed all instances of 'vhost' to
'server'.

The main reason for this is that NginX does not uphold the concept of
vHost: https://www.nginx.com/resources/wiki/start/topics/examples/server_blocks/

Stating:

```
"Note: “VirtualHost” is an Apache term. NGINX does not have Virtual
hosts, it has “Server Blocks” that use the server_name and listen
directives to bind to tcp sockets."
```

Besides, the extended use of the concept causes confusion and
misunderstanding when it comes to NginX.

I still need to work on two more requests.

### CODE REVIEW

**Comment by wyardley** [comment]
btw, I think the failing builds in Travis are due to hashrocket alignment. Try puppet-lint to see which lines.

**Comment by renich** [comment]
I've done some renaming. Maybe that was the cause.

puppet-lint shows a mess... unrelated. Namely, long lines and "not in autoload" errors.

**Comment by wyardley** [comment]
The long line warnings and 'not in autoload' ones can be ignored. `bundle exec rake release_checks` after doing the other prep stuff mentioned in the CONTRIBUTING file should give you a good idea of what things Travis will complain about

But if you look at the logs here:
https://travis-ci.org/voxpupuli/puppet-nginx/jobs/168808130

you can see that it's the hashrocket alignment that's at least one cause of the failing tests.

**Comment by renich** [comment]
Hah! Now ruby thinks it's python? ;D

Damn. I'll find some time tomorrow to work on this.

**Comment by renich** [comment]
OK, I never would've been able to sleep if I didn't try. Indentation issues fixed. ¬_¬... damn python...

File: manifests/resource/server.pp [hunk 1]
```diff
@@ -274,10 +274,10 @@
   $raw_append                   = undef,
   $location_raw_prepend         = undef,
   $location_raw_append          = undef,
-  $vhost_cfg_prepend            = undef,
-  $vhost_cfg_append             = undef,
-  $vhost_cfg_ssl_prepend        = undef,
-  $vhost_cfg_ssl_append         = undef,
+  $server_cfg_prepend            = undef,
```
Comment (bastelfreak) [inline manifests/resource/server.pp]: please adjust the = alignment here

**Comment by renich** [comment]
more indentation corrections.

**Comment by bastelfreak** [comment]
@renich can you have a look at the merge conflicts?

**Comment by renich** [comment]
@bastelfreak yep. I'll try to do it tomorrow... or something...

**Comment by renich** [comment]
OK, let's see if tests pass... I had to do it now...

**Comment by wyardley** [comment]
We didn't get to discuss this at the community modules meeting, but it's still on the slate, so should be able to discuss it next week. That said, if my recent PR that changes the structure pretty significantly gets merged, there will be a lot of conflicts (there are already some, but there will be more), so maybe hold off a bit on rebasing this until a decision gets made about some of the specifics of this cutover (i.e., do we provide a passthrough or a deprecation warning or not).

**Comment by renich** [comment]
ACK.

**Comment by wyardley** [comment]
This is going to happen (soonish), but I'm going to replace it with either #968 or its replacement. Welcome to follow / discuss in there.

**Pull request closed without merging**